cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -168,9 +168,14 @@ installCommand = CommandUI
       ++ "  " ++ pname ++ " v2-install ./pkgfoo\n"
       ++ "    Install the package in the ./pkgfoo directory\n"
 
-  , commandOptions      = nixStyleOptions clientInstallOptions
+  , commandOptions      = \x -> filter notInstallDirOpt $ nixStyleOptions clientInstallOptions x
   , commandDefaultFlags = defaultNixStyleFlags defaultClientInstallFlags
   }
+ where
+  -- install doesn't take installDirs flags, since it always installs into the store in a fixed way.
+  notInstallDirOpt x = not $ optionName x `elem` installDirOptNames
+  installDirOptNames = map optionName installDirsOptions
```

Comment:
Is this how command line flags are usually filtered out? I would think we would have plenty of of cases of command `foo` handling some flags that are really just dispatched to the underlying command `bar`?

Reply:
The issue is that nixStyleOptions _adds_ these things as options. So in this particular case we have nix style options which we specifically want to filter out from a particular v2 command.